sno/ogr_import_source.py

```diff
@@ -486,7 +486,34 @@ def _field_to_v2_column_schema(self, fd):
                     f"Unsupported column type for import: OGR subtype={ogr_subtype}"
                 )
 
-        if isinstance(data_type_info, tuple):
+        ogr_width = fd.GetWidth()
```

Comment:
data_type_info isn't used if this early-out happens - you should move this early-out to the very start of this function, otherwise the two flows are strangely interleaved.

Consider extracting this check into a new function - I am in favour of this since it's 30 lines long and has a different focus to the rest of the function - but most of it is comment, so maybe it doesn't make sense.
Something like:

if self._should_import_as_numeric(ogr_type, ogr_width):
  return ... (numeric type)